### PR TITLE
lib/errors: Add a test case against the error interface

### PR DESCRIPTION
**Pull request by indradhanush**
Also use t.Run for to make individual tests.

## Test plan

No functional change. Added a test. Build should pass.

### CODE REVIEW

File: lib/errors/warning_test.go [hunk 1]
```diff
@@ -8,24 +8,36 @@ func TestWarningError(t *testing.T) {
...
+			t.Error("Expected warning.As to return true but got false")
+		}
+	})
+
+	t.Run("test the warning.As method against the error interface.", func(t *testing.T) {
+		var e error
+		if w.As(e) {
+			t.Error("Expected warning.As to return false but got true")
+		}
+	})
```
Comment (indradhanush) [inline lib/errors/warning_test.go:36]: @bobheadxi Does this look like what you suggested in https://github.com/sourcegraph/sourcegraph/pull/38885#discussion_r936550684?

Comment (bobheadxi) [reply lib/errors/warning_test.go:36]: I meant the error interface we introduced a few days ago haha

Comment (bobheadxi) [reply lib/errors/warning_test.go:36]: https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/lib/errors/errors.go?L8:9-8:14

**Comment by jhchabran** [comment]
@indradhanush hey man, I'm going through GH notifs and stumbled across this one. You perhaps want to take action and forgot about it?

**Comment by indradhanush** [comment]
@jhchabran Oh I remember this now. I had to drop this for some other high prio work back then and forgot to come back to this. I will try to get it done this week or else close it.

**Comment by eseliger** [comment]
Closing as outdated.

**Pull request closed without merging**